src/simsopt/geo/boozersurface.py

```diff
@@ -104,20 +105,29 @@ def __init__(self, biotsavart, surface, label, targetlabel, constraint_weight=No
                 - `bfgs_maxiter` (int): maximum number of iterations for BFGS solver. Defaults to 1500.
                 - `limited_memory` (bool): True if L-BFGS solver is desired, False if the BFGS solver otherwise. Defaults to False.
                 - `weight_inv_modB` (float): for BoozerLS surfaces, weight the residual by modB so that it does not scale with coil currents.  Defaults to True.
+                - solver (int): list of flags to decide what algorithm(s) to use when running BoozerSurface.run_code. Only used if boozer_type is 'ls'. Default is ['ls', 'ls_newton']
```

Comment:
why only used if boozer_type is 'ls'?  I've used 'ls' first to get a good guess for 'exact'.  can we allow this use case?